code/go/0chain.net/chaincore/chain/protocol_block.go

```diff
@@ -379,7 +379,9 @@ func (c *Chain) finalizeBlock(ctx context.Context, fb *block.Block, bsh BlockSta
 
 	if len(fb.Events) > 0 && c.GetEventDb() != nil {
 		wg.Run("finalize block - add events", fb.Round, func() {
-			if err := c.GetEventDb().ProcessEvents(ctx, fb.Events, fb.Round, fb.Hash, len(fb.Txns)); err != nil {
+			if err := c.GetEventDb().ProcessEvents(
+				ctx, fb.Events, fb.Round, fb.Hash, len(fb.Txns), c.ChainConfig,
```

Comment:
Can we update the eventDb's config when c.ChainConfig is updated so that we don't have to check if need to update config on every ProcessEvents call